### PR TITLE
show number of active users

### PR DESCRIPTION
**Pull request by roxhens**
### Checklist
- [x] I tested *all* changes and *all* related features with different users (student, tutor, instructor, admin) on the test server https://artemistest.ase.in.tum.de.
- [x] Server: I added multiple integration tests (Spring) related to the features
- [x] Server: I added `@PreAuthorize` and check the course groups for all new REST Calls (security)
- [x] Server: I implemented the changes with a good performance and prevented too many database calls
- [x] Server: I documented the Java code using JavaDoc style.
- [x] Client: I added multiple integration tests (Jest) related to the features
- [x] Client: I added `authorities` to all new routes and check the course groups for displaying navigation elements (links, buttons)
- [x] Client: I documented the TypeScript code using JSDoc style.
- [x] Client: I added multiple screenshots/screencasts of my UI changes
- [x] Client: I translated all the newly inserted strings into German and English

### Motivation and Context

This change shows the number of active users in the application and is displayed in the application metrics section.

Issue #1330 

### Steps for Testing

1. Log in to Artemis
2. Navigate to Server Administration
3. Then go to Metrics
4. See the 'Active Users' field in the 'System' section.

### Screenshots

Screenshot 2020-05-03 at 17 07 59

### CODE REVIEW

**Comment by krusche** [comment]
Looks nice. Is this the same number as the current number of connected websocket connections?

One additional question: is there no way In Spring Boot to contribute this data to the existing metrics call?

**Comment by roxhens** [comment]
I guess is not the same as the current number of connected websocket connections, because I think that would show 2 connections if for example I am connected from my browser in my phone and in my pc with the same credentials, but the SimpUserRegistry shows 1.

According to the documentation this is the way Spring Boot tracks the connected users, so I thought this is the way to go. (link to [SimpUserRegistry](https://docs.spring.io/spring-framework/docs/current/javadoc-api/org/springframework/messaging/simp/user/SimpUserRegistry.html) doc.)

Regarding the second question: 
I thought this follows the same order as the other metrics, for example in the `metrics.service.ts` it makes GET requests to `/management/jhimetrics` and `management/threaddump` for System and Threads metrics respectively. 
But, maybe I can try to call and extend the `/management/jhimetrics` in a java service ,add the number of users to it and return it, so at the end there are 2 calls from Angular side.
Would this be better?

**Comment by krusche** [comment]
SimpUserRegistry should be fine. We also use this when logging the count of web socket users in `WebsocketConfiguration`.

I think it would be good if we can somehow add it to the jhimetrics call. It is defined in `https://github.com/jhipster/jhipster/blob/master/jhipster-framework/src/main/java/io/github/jhipster/config/metric/JHipsterMetricsEndpoint.java`

Please have a look if there is a way how to extend the data that is sent to the client in this REST call. Maybe it's also possible to extend what `MeterRegistry meterRegistry` is providing. Have a look at https://docs.spring.io/spring-metrics/docs/current/public/prometheus

File: src/main/java/de/tum/in/www1/artemis/web/rest/UserMetricsResource.java [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+     * GET - number of current users using the application.
+     * @return number of current users using the application.
+     */
+    @GetMapping("/usermetrics")
+    @PreAuthorize("hasAnyRole('ADMIN')")
+    public int getNumberOfUsers() {
+        return this.simpUserRegistry.getUserCount();
+    }
+
+}
```
Comment (jpbernius) [inline src/main/java/de/tum/in/www1/artemis/web/rest/UserMetricsResource.java:29]: From my understanding, this is unused and can go away?
```suggestion
```

Comment (roxhens) [reply src/main/java/de/tum/in/www1/artemis/web/rest/UserMetricsResource.java:29]: Yes, that is correct. I removed it now. Thank you.

**Pull request merged**